src/solidity-frontend/solidity_grammar.cpp

```diff
@@ -426,7 +491,35 @@ ExpressionT get_expression_t(const nlohmann::json &expr)
   return ExpressionTError;
 }
 
-ExpressionT get_expr_operator_t(const nlohmann::json &expr, bool uo_pre)
+ExpressionT get_unary_expr_operator_t(const nlohmann::json &expr, bool uo_pre)
+{
+  if(expr["operator"] == "--")
+  {
+    if(uo_pre)
+      return UO_PreDec;
+
+    assert(!"Unsupported - UO_PostDec");
+  }
+  if(expr["operator"] == "++")
+  {
+    if(uo_pre)
+      return UO_PreInc;
+
+    assert(!"Unsupported - UO_PostDec");
+  }
+  if(expr["operator"] == "-")
+  {
+    return UO_Minus;
+  }
+  assert(!((fmt::format(
+              "Got expression operator={}. Unsupported expression operator",
+              expr["operator"].get<std::string>()))
+             .c_str()));
+
+  return ExpressionTError; // make some old compilers happy
```

Comment:
I don't understand. You're asserting `false` just above and now `make some old compilers happy`? What compilers are meant here? Also new compilers will happily complain if this function doesn't return anything despite the `assert`. Asserts get compiled out on `Release` builds, old and new compilers will not know what to do then.

Either
- leave as it is without the comment, or
- print the error message and `abort()` - old and new compilers will be happy, or
- use `__builtin_unreachable` (least preferrable option)

Reply:
Thanks!